### PR TITLE
Implement Symbol:isBinaryOp and Symbol:isIdentifier

### PR DESCRIPTION
**Pull request by snappizz**
implement #2952. we're going to be using these methods in SCDoc so why not make 'em faster with primitives. also, i need some experience writing primitives.

all documented, unit tested, and ready to go.

### CODE REVIEW

**Comment by snappizz** [comment]
sorry, what?

**Comment by telephon** [comment]
sorry, I misread. Never mind!

File: lang/LangPrimSource/PyrSymbolPrim.cpp [hunk 1]
```diff
@@ -179,6 +179,82 @@ int prSymbolIsMetaClassName(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+const char *binary_op_characters = "!@%&*-+=|<>?/";
```
Comment (brianlheim) [inline lang/LangPrimSource/PyrSymbolPrim.cpp]: Sorry, should have been clearer—you can put them inside the function `prSymbol_IsBinaryOp` with `static` scope, and that means they are treated as init-once storage within the function but with local scope.

Comment (snappizz) [inline lang/LangPrimSource/PyrSymbolPrim.cpp]: ok, fixed

**Pull request merged**